source/Options.cpp

```diff
@@ -147,336 +147,176 @@ Options::Options(
       export_origins_mode_(export_origins_mode),
       propagate_across_arguments_(propagate_across_arguments) {}
 
-Options::Options(const boost::program_options::variables_map& variables) {
-  system_jar_paths_ = parse_paths_list(
-      variables["system-jar-paths"].as<std::string>(),
-      std::nullopt,
-      /* check exist */ false);
+Options::Options(const Json::Value &json){
+    system_jar_paths_ = parse_paths_list(
+        json["system-jar-paths"].asString(),
+        std::nullopt,
+        /* check exist */ false);
 
-  apk_directory_ =
-      check_directory_exists(variables["apk-directory"].as<std::string>());
-  dex_directory_ =
-      check_directory_exists(variables["dex-directory"].as<std::string>());
+    apk_directory_ =
+        check_directory_exists(json["apk-directory"].asString());
```

Comment:
Here and many other places: you can use JsonValidation::string like this:
```suggestion
    apk_directory_ =
        check_directory_exists(JsonValidation::string(json, "apk-directory"));
```